packages/js/src/preferences/preferences.ts

```diff
@@ -1,3 +1,32 @@
 import { BaseModule } from '../base-module';
```

Comment:
For a future PR: The naming is a bit off. Is this a `BaseModule` or `BaseResource`, or `BaseModel`?

Reply:
yes I was thinking what could be a better name here, what you would suggest instead?